network/src/tests/receiver_tests.rs

```diff
@@ -49,3 +74,81 @@ async fn receive() {
     let received = message.unwrap();
     assert_eq!(received, sent);
 }
+
+#[tokio::test]
+async fn receive_no_ack() {
+    // Make the network receiver.
+    let address = "127.0.0.1:4001".parse::<SocketAddr>().unwrap();
+    let (tx, mut rx) = channel(1);
+    Receiver::spawn(address, TestHandlerNoAck { deliver: tx });
+    sleep(Duration::from_millis(50)).await;
+
+    // Send a message.
+    let sent = "Hello, world!";
+    let bytes = Bytes::from(bincode::serialize(sent).unwrap());
+    let stream = TcpStream::connect(address).await.unwrap();
+    let mut transport = Framed::new(stream, LengthDelimitedCodec::new());
+    transport.send(bytes.clone()).await.unwrap();
+
+    // Ensure the message gets passed to the channel.
+    let message = rx.recv().await;
+    assert!(message.is_some());
+    let received = message.unwrap();
+    assert_eq!(received, sent);
+}
+
+#[tokio::test]
+async fn accounting_receive() {
```

Comment:
I would suggest to extract this to its own dedicated file `accounting_receiver_tests` so its clearer to expect where to find the component's tests